### ISSUE TITLE
`aircraft_mass` shape is incompatible with `true_airspeed`

### ISSUE THREAD

**Issue by jiaweikho** [contrailcirrus/pycontrails#246]
**Description**

When `aircraft_mass` is fetched from `Flight.attrs`, the array shape is not compatible with the `true_airspeed` shape when evaluating `pycontrails.models.cocip.Cocip_simulate_wake_vortex_downwash()`. The shapes of `aircraft_mass` and `true_airspeed` have diverged after applying the Schmidt-Appleman Criterion filter in `pycontrails.models.cocip.Cocip._find_initial_contrail_regions()`. The filter is applied to `_sac_flight` but not to any of the attributes. Perhaps this is because there is no distinction between attributes that specific to the flight configuration and those that are time-varying like `aircraft_mass`. It is probably also important to note that this is only a problem when a `Flight` object is not instantiated from a DataFrame and instead constructed from the keywords and an `attr` dictionary.

**Details**

- Version: 0.46.1
- Module: pycontrails.models.cocip
- OS: ubuntu

**Steps to Reproduce**

1. I don't have a repository to reproduce this from but you can use the script outline in the [Run CoCiP over a Flight](https://py.contrails.org/notebooks/run-cocip-on-flight.html) tutorial with the exception that the `Flight` object is instantiated like:

```python
fl = Flight(
    time=df["time"],
    latitude=df["latitude"],
    longitude=df["longitude"],
    altitude=df["altitude"],
    attrs={
        **attrs,
        "aircraft_mass": df["aircraft_mass"],
    },
```
2. Evaluate the flight as per the rest of the tutorial.

**Additional Notes**

- I have v0.46.1 installed locally but I believe this is still an issue on `main`.
- If `aircraft_mass` is required for the CoCiP model, I think this could be solved by adding an "aircraft_mass" keyword to the Flight constructor to provide the option in the same way the option is provided for "time", "latitude", "longitude", "altitude". This should mean that the aircraft_mass array is also filtered when applying the SAC.

**Comment by zebengberg** [comment]
Thanks for describing your issue so clearly.

The `Flight.attrs` was designed to hold arbitrary properties that are constant over the entire. Time-varying quantities, like aircraft mass, should be included in `Flight.data`.

For your particular case, you have a few easy options:

1. Pass the `df` directly into the `Flight` constructor. This might be the easiest approach.

```python
fl = Flight(df, attrs=attrs)
```

You could also select a subset of the columns before passing it in.

```python
fl = Flight(df[["longitude", "latitude", "altitude", "time", "aircraft_mass"]], attrs=attrs)
```

2. Add the aircraft mass after instantiation:

```python
fl = Flight(
    time=df["time"],
    latitude=df["latitude"],
    longitude=df["longitude"],
    altitude=df["altitude"],
    attrs=attrs,
)
fl["aircraft_mass"] = df["aircraft_mass"]
```

**Comment by jiaweikho** [comment]
Hi @zebengberg, I think you're exactly right, both solutions you've proposed would work but since the instantiation of Flight is very flexible, it is not immediately obvious that the `aircraft_mass` should be declared in the DataFrame instead. When the `Flight` object is instantiated without the `aircraft_mass` attribute, a `ValueError` is thrown from:
https://github.com/contrailcirrus/pycontrails/blob/7602a4d933cbffdc8295e7d17b797da0f17b9756/pycontrails/models/cocip/cocip.py#L1611-L1646

From this information, all other `ap_vars` are declared in `flight.attr` so it seemed logical to include `aircraft_mass` in `flight.attr` too. Perhaps this is just a matter of improving the documentation to outline where `aircraft_mass` should be declared?

Is there a reason time-varying properties like "time", "latitude", "longitude", and "altitude" are available as keyword arguments despite also being required in the `data` field? Why was this not available to `aircraft_mass` too?

**Comment by zebengberg** [comment]
Yep, the pycontrails documentation can definitely be improved! If you're up for it, please help expand and open a PR.

Adding the temporal-spatial fields as keyword arguments may seem like arbitrary choice. The motivation was that every `Flight` instance must contain these four temporal-spatial fields, whereas aircraft performance quantities (like aircraft mass, or engine efficiency, or fuel flow) are not required. Some models (such as `Cocip`) require additional data, like the aircraft performance estimates. If known, these can be included in the `Flight` object. If not known, you can estimate them with an aircraft performance model such as `PSFlight`. We have [some documentation](https://py.contrails.org/notebooks/AircraftPerformance.html) discussing this in more detail.
